internal/terraform/node_resource_plan_instance.go

```diff
@@ -202,28 +206,19 @@ func (n *NodePlannableResourceInstance) managedResourceExecute(ctx EvalContext)
 		}
 	}
 
+	var refreshDiags tfdiags.Diagnostics
```

Comment:
nit: push the creation of the refreshDiags into the if statement, just so it's not confusing the larger function scope.